server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -392,6 +392,9 @@ public void setAliasFilter(AliasFilter aliasFilter) {
     }
 
     public void source(SearchSourceBuilder source) {
+        if (source != null) {
+            source = source.minimalSourceForShardRequests();
```

Comment:
Should we call this on the constructor too ?

Reply:
+1